package-res/resources/web/vizapi/DataTable.js

```diff
@@ -15,1094 +15,1220 @@
 *
 */
 
-/*
-pentaho.DataTable
-pentaho.DataView
+pentaho = typeof pentaho == "undefined" ? {} : pentaho;
 
-Client-side non-visual data tables.
+/**
+ * Pentaho JavaScript Visualization Data APIs.
+ *
+ * This module defines interfaces and classes
+ * used by visualizations for data representation and manipulatation.
+ *
+ * @module common-ui.vizapi.data
+ * @main common-ui.vizapi.data
+ */
+
+/**
+ * An _AbstractDataTable_ represents a set of tabular data.
+ *
+ * @class AbstractDataTable
+ * @constructor
+ */
+ pentaho.AbstractDataTable = function AbstractDataTable() {};
+
+/**
+ * Gets a column object, given its index.
+ *
+ * @method _getColumn
+ * @param {Number} colIdx The column index (zero-based).
+ * @return {Object} The column object.
+ * @protected
+ */
+
+/**
+ * Gets a row object, given its index.
+ *
+ * @method _getRow
+ * @param {Number} rowIdx The row index (zero-based).
+ * @return {Object} The row object.
+ * @protected
+ */
+
+/**
+ * Gets a cell, given its row and column index.
+ *
+ * @method _getCell
+ * @param {Number} rowIdx The cell's row index (zero-based).
+ * @param {Number} colIdx The cell's column index (zero-based).
+ * @return {Null|Object} The cell.
+ * @protected
+ */
+
+/**
+ * Gets the number of columns.
+ *
+ * @method getNumberOfColumns
+ * @return {Number} The number of columns.
+ */
+
+/**
+ * Gets the number of rows.
+ *
+ * @method getNumberOfRows
+ * @return {Number} The number of rows.
+ */
+
+// TODO: Not publicizing the add method yet, as it is not unit-tested,
```

Comment:
Suggest removing the TODOs. Not sure if it needs to be fixed in this pull request. If they are documented in another Backlog item then the TODO is probably not needed. If it is working as expected maybe remove the TODO part and just have a comment of current limitations.